src/observables/ForkJoinObservable.ts

```diff
@@ -1,50 +1,96 @@
 import {Observable} from '../Observable';
 import {Subscriber} from '../Subscriber';
+import {PromiseObservable} from './PromiseObservable';
+import {EmptyObservable} from './EmptyObservable';
+import {isPromise} from '../util/isPromise';
 
 export class ForkJoinObservable<T> extends Observable<T> {
-  constructor(private observables: Observable<any>[]) {
-    super();
+  constructor(private sources: Array<Observable<any> |
+                                    Promise<any> |
+                                    ((...values: Array<any>) => any)>) {
+     super();
+   }
```

Comment:
I don't mind it at all, actually I also do similar thing when I review with my colleagues :) I'll check codes and apply changes if necessary.